### DISCUSSION TITLE
导出csv文件很慢

### DISCUSSION THREAD

**Discussion by heihei180** [hhyo/Archery#2620] [General]
最近有几次处理生产数据，需要将17W数据导出，尝试了几次后，发现浏览器提示oom直接奔溃，后来选择每次导出3-4W条，在手动将csv文件合并到一起，但即便是这样，每次3-4W 一次也得20多分钟，是否有优化的计划呢？

我执行的sql很简单：

```sql
select id, mn, pn, cpn from mif where pn= 'ZTxxxx' AND DELETE_FLAG = 0
and id>= 6300000 and   id< 6400000
```

**Comment by LeoQuote** [comment]
当前导出是前端提供的功能, 欢迎 pr

**Reply by heihei180** [reply]
啊？难怪，我说python慢，怎么这么慢，也不应该造成浏览器oom呀，我看下代码，如果可以的话，我提一个PR

**Reply by LeoQuote** [reply]
本项目开发文档如下, 希望对你有帮助:

* https://github.com/hhyo/Archery/wiki/development
* https://github.com/hhyo/Archery/wiki/notifier

期待你的pr~

**Reply by heihei180** [reply]
:(  实在抱歉，我看了下，这块代码都是bootstrap，我不太了解bootstrap，有些看不懂。。。不怎么敢改了

**Reply by LeoQuote** [reply]
你可以新建一个审批流从后台导，以临时下载链接，邮件发送等方式提供给用户

**Reply by heihei180** [reply]
好的，谢谢 我尝试一下
